### PR TITLE
QA-378: Add AUTOMATION tests for Downloads and mender-client deb install

### PR DESCRIPTION
**Pull request by lluiscampos**
* Install with Debian: Use native arch and split download and setup
* Install with Debian: Use new demo flags on mender setup commands
* Downloads: Consistently use sudo in all privileged commands
* Downloads: fix download links for mender-gateway examples package
* Downloads: fix download links for mender-gateway Debian buster armhf
* Downloads: use wget for mender-gateway examples package
* Downloads: Split closed source downloads instructions
* QA-378: Add AUTOMATION tests for Downloads and mender-client deb install

### CODE REVIEW

**Comment by lluiscampos** [comment]
@TheYoctoJester please review all except the last commit

@kacf @oleorhagen review all :)

**Comment by lluiscampos** [comment]
Pipeline from protected branch running enterprise tests [here](https://gitlab.com/Northern.tech/Mender/mender-docs/-/pipelines/499155207).

File: 03.Client-installation/02.Install-with-Debian-package/docs.md [hunk 1]
```diff
@@ -26,18 +26,19 @@ architectures:
...
-this is the most common for users getting starting with Mender.
+package architectures.
 
 
 ### Download the package
 
 <!--AUTOVERSION: "downloads.mender.io/%/"/mender "mender-client_%-1"/mender -->
 ```bash
-wget https://downloads.mender.io/3.1.1/dist-packages/debian/armhf/mender-client_master-1%2Bdebian%2Bbuster_armhf.deb
+wget https://downloads.mender.io/3.1.1/dist-packages/debian/$(dpkg --print-architecture)/mender-client_master-1%2Bdebian%2Bbuster_$(dpkg --print-architecture).deb
```
Comment (TheYoctoJester) [inline 03.Client-installation/02.Install-with-Debian-package/docs.md]: why 3.1.1?

File: 03.Client-installation/02.Install-with-Debian-package/docs.md [hunk 2]
```diff
@@ -26,18 +26,19 @@ architectures:
 - amd64: Generic 64-bit x86 processors, the most popular among workstations
 
 See [the downloads page](../../09.Downloads/docs.md) for links to download all
-package architectures. We will assume *armhf* in the following instructions as
-this is the most common for users getting starting with Mender.
+package architectures.
```
Comment (TheYoctoJester) [inline 03.Client-installation/02.Install-with-Debian-package/docs.md:29]: It would be nice to have a short list of provided architectures.

Comment (lluiscampos) [reply 03.Client-installation/02.Install-with-Debian-package/docs.md:29]: The list is few lines above. Isn't that enough?

File: 03.Client-installation/02.Install-with-Debian-package/docs.md (hunk 1 above)
Comment (lluiscampos) [reply 03.Client-installation/02.Install-with-Debian-package/docs.md]: That was a (pre-existing) mistake and the download link was broken. I corrected it in the last commit, for the sake of keeping the commits comprehensive I didn't fixed it when replacing the hardcoded architectures with `dpkg --print-architecture`

File: 03.Client-installation/02.Install-with-Debian-package/docs.md [hunk 3]
```diff
@@ -63,49 +64,70 @@ device is now ready to authenticate with the server and start receiving updates.
 Alternatively, install the package non-interactively. This is suitable for
 scripts or other situations where no user input is desired.
 
+First, to install Mender without configuring it run the following command:
```
Comment (oleorhagen) [inline 03.Client-installation/02.Install-with-Debian-package/docs.md:70]: All nice :ok_hand: 

Maybe we should tab this as well? I mean, it's basically the same 3x times

File: 03.Client-installation/02.Install-with-Debian-package/docs.md [hunk 4]
```diff
@@ -31,9 +31,11 @@ package architectures.
 
 ### Download the package
 
+<!-- AUTOMATION: execute=DEBIAN_FRONTEND=noninteractive apt-get install -y libglib2.0-0 tzdata -->
+
 <!--AUTOVERSION: "downloads.mender.io/%/"/mender "mender-client_%-1"/mender -->
 ```bash
-wget https://downloads.mender.io/3.1.1/dist-packages/debian/$(dpkg --print-architecture)/mender-client_master-1%2Bdebian%2Bbuster_$(dpkg --print-architecture).deb
+wget https://downloads.mender.io/master/dist-packages/debian/$(dpkg --print-architecture)/mender-client_master-1%2Bdebian%2Bbuster_$(dpkg --print-architecture).deb
```
Comment (oleorhagen) [inline 03.Client-installation/02.Install-with-Debian-package/docs.md:38]: :+1:

File: 09.Downloads/docs.md [hunk 5]
```diff
@@ -8,6 +8,8 @@ process:
     twig: true
 ---
 
+<!-- AUTOMATION: execute=if [ "$TEST_ENTERPRISE" != 1 ]; then echo "TEST_ENTERPRISE must be set to 1!"; exit 1; fi -->
```
Comment (oleorhagen) [inline 09.Downloads/docs.md:11]: Q:
 [ ... `-ne` 1 ] ?

Comment (oleorhagen) [reply 09.Downloads/docs.md:11]: Or quote the `1`?

File: 09.Downloads/docs.md [hunk 6]
```diff
@@ -169,9 +177,11 @@ repository. Afterwards, you can install and update the Mender client using the
 !!! With this method the latest released Mender components will be installed
 
 1. Update the `apt` package index and install required dependencies.
+
+<!-- AUTOMATION: execute=DEBIAN_FRONTEND=noninteractive apt-get install -y tzdata -->
    ```bash
    sudo apt-get update
-   sudo apt-get install \
+   sudo apt-get install --assume-yes \
```
Comment (oleorhagen) [inline 09.Downloads/docs.md:184]: Testing has it's benefits :smile_cat:

File: 09.Downloads/docs.md [hunk 7]
```diff
@@ -412,8 +441,7 @@ Then install the package with:
 
 <!--AUTOVERSION: "mender-monitor_%-1"/monitor-client -->
 ```bash
-sudo dpkg -i mender-monitor_master-1+debian+buster_all.deb
-sudo apt --fix-broken -y install
+sudo dpkg -i mender-monitor_master-1+debian+buster_all.deb || apt --fix-broken -y install
```
Comment (oleorhagen) [inline 09.Downloads/docs.md:444]: `sudo apt fix` as well, right?

File: run-tests.sh [hunk 8]
```diff
@@ -4,10 +4,12 @@ set -ex
...
 ./test_autoversion.py
 ./autoversion.py --check
+./test_docs.py 03.Client-installation/02.Install-with-Debian-package/docs.md
 env TEST_OPEN_SOURCE=1 ./test_docs.py 07.Server-installation/03.Installation-with-docker-compose/docs.md
 if [ -n "$REGISTRY_MENDER_IO_PASSWORD" ]; then
     docker login -u ntadm_menderci -p "$REGISTRY_MENDER_IO_PASSWORD" registry.mender.io
     env TEST_ENTERPRISE=1 ./test_docs.py 07.Server-installation/03.Installation-with-docker-compose/docs.md
     env TEST_ENTERPRISE=1 ./test_docs.py 07.Server-installation/03.Installation-with-docker-compose/02.Upgrading-from-OS-to-Enterprise/docs.md
     env TEST_ENTERPRISE=1 ./test_docs.py 08.Server-integration/03.Mutual-TLS-authentication/docs.md
+    env TEST_ENTERPRISE=1 ./test_docs.py 09.Downloads/docs.md
```
Comment (oleorhagen) [inline run-tests.sh:14]: :muscle:

**Review by oleorhagen — APPROVED** [review]
Thanks, logical, structured commits.

Was a breeze to review :100:

File: 09.Downloads/docs.md (hunk 7 above)
Comment (kacf) [inline 09.Downloads/docs.md:444]: Missing `sudo` on the last command.

**Review by kacf — APPROVED** [review]
Very nice testing! Pointed out a couple of small things, but pre-approving.

File: 09.Downloads/docs.md [hunk 9]
```diff
@@ -244,11 +261,16 @@ repository. Afterwards, you can install and update the Mender client using the
...
 4. Update the package index and install the Mender client:
+
+<!--AUTOMATION: ignore -->
    ```bash
    sudo apt-get update
    sudo apt-get install mender-client
    ```
 
+<!-- AUTOMATION: execute=apt-get update -->
+<!-- AUTOMATION: execute=DEBIAN_FRONTEND=noninteractive apt-get install mender-client -->
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply 09.Downloads/docs.md:272]: Ok, thanks.

File: 09.Downloads/docs.md (hunk 5 above)
Comment (lluiscampos) [reply 09.Downloads/docs.md:11]: I'll fix this in all files :+1:

File: 03.Client-installation/02.Install-with-Debian-package/docs.md (hunk 3 above)
Comment (lluiscampos) [reply 03.Client-installation/02.Install-with-Debian-package/docs.md:70]: Nice suggestion! Done

**Comment by lluiscampos** [comment]
@oleorhagen I appreciate your eyes on the last two commits - just to make sure I haven't done anything stupid.

**Comment by oleorhagen** [comment]
:green_circle: @lluiscampos

**Pull request merged**